tracker-db/openshift/templates/tracker-db.yaml

```diff
@@ -0,0 +1,270 @@
+  - kind: Secret
+    apiVersion: v1
+    metadata:
+      name: ${NAME}-${TAG}-secret
+      labels:
+        name: ${NAME}
+        environment: ${TAG}
+        role: ${ROLE}
+    stringData:
+      database-name: ${POSTGRESQL_DATABASE_NAME}
+      database-user: ${POSTGRESQL_USER}
+      database-password: ${POSTGRESQL_PASSWORD}
+      admin-password: ${POSTGRESQL_ADMIN_PASSWORD}
+    type: Opaque
+
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: ${NAME}-${TAG}
+      labels:
+        name: ${NAME}
+        environment: ${TAG}
+        role: ${ROLE}
+      annotations:
+        template.openshift.io/expose-uri: postgres://{.spec.clusterIP}:{.spec.ports[?(.name==postgresql)].port}
+    spec:
+      ports:
+        - name: ${NAME}-${TAG}
+          protocol: TCP
+          port: 5432
+          targetPort: 5432
+      selector:
+        name: ${NAME}
+        environment: ${TAG}
+      type: ClusterIP
+      sessionAffinity: None
+
+parameters:
+  - name: NAME
+    displayName: Name
+    description:
+      The name assigned to all of the OpenShift resources associated to the
+      PostgreSQL instance.
+    required: true
+    value: postgresql
+
+  - name: TAG
+    displayName: Environment TAG name
+    description: The TAG name for this environment, e.g., dev, test, prod
+    required: true
+    value: dev
+
+  - name: ROLE
+    displayName: Role
+    description: Role
+    required: true
+    value: db
+
+  - name: IMAGE_NAMESPACE
+    displayName: Image Namespace
+    required: true
+    description: The namespace of the OpenShift project containing the imagestream for the application.
+    value: openshift
+
+  - name: SOURCE_IMAGE
+    displayName: Source Image name
+    required: true
+    description: The namespace of the OpenShift project containing the imagestream for the application.
+    value: postgresql:12
+
+  - name: POSTGRESQL_DATABASE_NAME
+    displayName: PostgreSQL Database Name
+    description: The name of the PostgreSQL database.
+    required: true
+    value: tracker
+
+  - name: POSTGRESQL_USER
+    displayName: PostgreSQL Connection Username
+    description: Username for PostgreSQL user that will be used for accessing the database.  Needs to be basee64 encoded/
+    required: true
+    value: user5SJ
+
+  - name: POSTGRESQL_PASSWORD
+    displayName: PostgreSQL Connection Password
+    description: Password for the PostgreSQL connection user.  Needs to be basee64 encoded/
+    required: true
+    generate: expression
+    from: '[a-zA-Z0-9]{16}'
+
+  - name: POSTGRESQL_ADMIN_PASSWORD
+    displayName: PostgreSQL Admin Password
+    description: Password for the 'postgres' PostgreSQL administrative account.  Needs to be basee64 encoded.
+    required: true
+    generate: expression
+    from: '[a-zA-Z0-9]{16}'
+
+  - name: PERSISTENT_VOLUME_SIZE
+    displayName: Persistent Volume Size
+    description: The size of the persistent volume , e.g. 512Mi, 1Gi, 2Gi.
+    required: true
+    value: 10Gi
```

Comment:
Need to review the size of PVC. The namespace may not have enough space for it.

Reply:
Need confirmation on what value should be used for this.  Have left as is for now.

Reply:
Currently, lear db and backup requested total  30G space. It has 20G leave over. You may need to estimate your database size for db and backup.

Reply:
hmmmm...it's only one table but it will grow over time as it tracks completed messages too.  Ideally we would remove completed messages after a certain threshold.  Let's say we start with 15GB for tracker_db and its backup.  If that's the case, do you need to do anything additional to support this or can I just update to 15Gi in the above yaml section?

Reply:
As per our conversation, this has now been updated to 5G.  

The database is relatively small and inserting 1000 test records came out to be 560kB.  Extrapolating that to 2 years gives us 409 MBs.

Reply:
cool